pkg/provider/vault/vault.go

```diff
@@ -245,7 +246,7 @@ func (c *Connector) newClient(ctx context.Context, store esv1beta1.GenericStore,
 }
 
 func (c *Connector) NewGeneratorClient(ctx context.Context, kube kclient.Client, corev1 typedcorev1.CoreV1Interface, vaultSpec *esv1beta1.VaultProvider, namespace string) (util.Client, error) {
-	vStore, cfg, err := c.prepareConfig(kube, corev1, vaultSpec, namespace, "Generator")
+	vStore, cfg, err := c.prepareConfig(kube, corev1, vaultSpec, nil, namespace, "Generator")
```

Comment:
I'm not too sure if we can or should pass `retrySettings` here, so I need advise on this point from others 🙇

Reply:
The generator spec at https://external-secrets.io/v0.8.5/api/generator/vault/ doesn't have it, nor do the other generators.

If it's desired it should IMHO be a separate PR, that I may be able to complete myself.